tensordict/prototype/tensorclass.py

```diff
@@ -238,7 +230,6 @@ def wrapper(self, key, value):
         if type(value) in CLASSES_DICT.values():
             value = value.__dict__["tensordict"]
         self.__dict__["tensordict"][key] = value
-        assert self.__dict__["tensordict"][key] is value
```

Comment:
This check was failing both for `None` values, but also tensors that were moved to a new device when assigned to the tensordict, so I decided to just delete it. lmk though if you think we need to retain some kind of input validation.

Reply:
I agree with you. we can drop that check.

Reply:
+ we don't want `assert` in the code base, these should be kept for tests only